charts/redpanda/values_util.go

```diff
@@ -0,0 +1,60 @@
+package redpanda
+
+import (
+	"fmt"
+
+	"github.com/invopop/jsonschema"
+)
+
+type ImageTag string
+
+func (ImageTag) JSONSchemaExtend(schema *jsonschema.Schema) {
+	schema.Pattern = `^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$|^$`
+}
+
+type ImageRepository string
+
+func (ImageRepository) JSONSchemaExtend(schema *jsonschema.Schema) {
+	schema.Pattern = "^[a-z0-9-_/.]+$"
+}
+
+type MemoryAmount string
+
+func (MemoryAmount) JSONSchemaExtend(schema *jsonschema.Schema) {
+	schema.Pattern = "^[0-9]+(\\.[0-9]){0,1}(k|M|G|Ki|Mi|Gi)$"
+}
+
+type IssuerRefKind string
+
+func (IssuerRefKind) JSONSchemaExtend(schema *jsonschema.Schema) {
+	schema.Enum = append(schema.Enum, "ClusterIssuer", "Issuer")
+}
+
+type Enablable struct {
```

Comment:
do you mean Enable?

Reply:
No, `Enabled` is the field within it. This is a generic type for anything that's shaped like `{enabled: true}` which I'm calling an "enablable". I'll update the comments before merging!

Reply:
Eh, I decided to just replaced these with inline structs for now. It's more clear that way. We can always swap back or start using embedding if desired.